google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/RetryInfoTest.java

```diff
@@ -368,6 +545,45 @@ private void verifyRetryInfoCanBeDisabled(Runnable runnable) {
     assertThat(attemptCounter.get()).isEqualTo(1);
   }
 
+  // Test the case where server does not return retry info
+  private void verifyNoRetryInfo(Runnable runnable, boolean operationRetryable) {
+    enqueueRetryableExceptionNoRetryInfo();
+
+    if (!operationRetryable) {
+      try {
+        runnable.run();
+        Assert.fail("non retryable operation should fail");
+      } catch (ApiException e) {
+        assertThat(attemptCounter.get()).isEqualTo(1);
+      }
+    } else {
+      Stopwatch stopwatch = Stopwatch.createStarted();
+      runnable.run();
+      stopwatch.stop();
+
+      assertThat(attemptCounter.get()).isEqualTo(2);
+      assertThat(stopwatch.elapsed()).isLessThan(Duration.ofSeconds(delay.getSeconds()));
+    }
+
+    attemptCounter.set(0);
+
+    ApiException exception = enqueueNonRetryableExceptionNoRetryInfo();
+
+    try {
```

Comment:
similar to above:

ApiException e = assertThrows(ApiException.class, runnable);

// Extract the underlying error from the first entry for batch mutations
if (e instanceof MutateRowsException) {
  FailedMutation firstEntryResult = ((MutateRowsException) e).getFailedMutations().get(0);
  e = firstEntryResult.getError();
}

assertThat(e.getStatusCode()).isEqualTo(...)